src/main/java/software/amazon/awssdk/crt/CRT.java

```diff
@@ -149,13 +150,50 @@ private static void extractAndLoadLibrary(String path) {
             }
 
             // Prefix the lib
-            String prefix = "AWSCRT_" + new Date().getTime();
+            String prefix = "AWSCRT_";
             String libraryName = System.mapLibraryName(CRT_LIB_NAME);
-            String libraryPath = "/" + getOSIdentifier() + "/" + getArchIdentifier() + "/" + libraryName;
 
             File tempSharedLib = File.createTempFile(prefix, libraryName, tmpdirFile);
 
+			// The temp lib file should be deleted when we're done with it.
+			// Ask Java to try and delete it on exit. We call this immediately
+			// so that if anything goes wrong writing the file to disk, or
+			// loading it as a shared lib, it will still get cleaned up.
+			tempSharedLib.deleteOnExit();
+
+            // Unfortunately File.deleteOnExit() won't work on Windows, where
+            // files cannot be deleted while they're in use. On Windows, once
+            // our .dll is loaded, it can't be deleted by this process.
+            //
+            // The Windows-only solution to this problem is to scan on startup
+            // for old instances of the .dll and try to delete them. If another
+            // process is still using the .dll, the delete will fail, which is fine.
+            String os = getOSIdentifier();
+            if (os == "windows") {
```

Comment:
Feel like platform id shouldn't be a string, but that's a larger change.

Reply:
.equals()?

Reply:
Another great reason to make it not-a-string!

Reply:
I started making it not-a-string but it was turning into a big refactor so going with `.equals()` as suggested by Dombo